SpatialGDK/Documentation/content/non-unreal-layers.md

```diff
@@ -0,0 +1,213 @@
+# Non-Unreal layers
+<%(TOC)%>
+
+Having multiple layers in a SpatialOS game means using multiple worker types to simulate different aspects of the same deployment. For example, your game could use Unreal servers workers developed with the GDK *and* a separate Python Database worker built using the [C API](https://docs.improbable.io/reference/latest/capi/introduction) in coordination with a 3rd party service. This lets you modularize game elements, and reuse these separate worker types across multiple games.
+
+In order to interact with other layers, the Unreal GDK needs to send data, receive data, and edit snapshots, for components defined in external schema files. This functionality is made possible through direct use of the C API within the Unreal GDK.
+
+## How multi-layer is implemented
+
+Sending data directly via the C API can be achieved using methods defined in the `SpatialWorkerConnection` (described in the section below). Receiving network ops for externally defined Schema components, which do not have Unreal component equivalents, is not handled by default as part of GDK pipeline. Instead, you can provide custom callbacks for specific component IDs and ops, and the GDK pipeline forwards the ops to your methods. Similarly, you can provide custom snapshot generation functions letting you serialize and add additional entities to the Snapshot output stream, which runs when the `Snapshot` button in the toolbar is clicked.
+
+## How to use multi-layer
+
+### Sending data
+
+You can send component updates, command requests, and command responses directly to the C API using the `SpatialWorkerConnection.h` public methods:
+
+```
+void SendComponentUpdate(Worker_EntityId EntityId, const Worker_ComponentUpdate* ComponentUpdate);
+Worker_RequestId SendCommandRequest(Worker_EntityId EntityId, const Worker_CommandRequest* Request, uint32_t CommandId);
+void SendCommandResponse(Worker_RequestId RequestId, const Worker_CommandResponse* Response);
+```
+
+which are accessible via a reference to the net driver:
+
+```
+SpatialWorkerConnection* Connection = Cast<USpatialNetDriver>(World->GetNetDriver())->Connection;
+```
+
+There is a basic example in the section below, and for more examples of how to construct component updates, command requests, and more, check the [C API page on serialization](https://docs.improbable.io/reference/latest/capi/serialization).
+
+### Receiving data
+
+You can receive components updates through implementing dispatcher callbacks for specified component IDs and op types. These callbacks are detected by the Unreal dispatcher pipeline on initialization and used instead of the existing pipeline where the component ID and op type are matched.
```

Comment:
```suggestion
You can receive component updates through implementing dispatcher callbacks for specified component IDs and op types. These callbacks are detected by the Unreal dispatcher pipeline on initialization and used instead of the existing pipeline where the component ID and op type are matched.
```